packages/l10n/src/create-l10n-injector/create-l10n-injector.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { reportErrorToSentry } from '@commercetools-frontend/sentry';
+import { getDisplayName } from '../utils';
+
+type InjectorOptions<T> = {
+  displayName: string;
+  propKey: string;
+  propLoadingKey: string;
+  loadLocale: (locale: string, cb: (error?: Error, data?: T) => void) => void;
+};
+
+export default function createL10NInjector<T>({
+  displayName,
+  propKey,
+  propLoadingKey,
+  loadLocale,
+}: InjectorOptions<T>) {
+  return function createHOC<U>(mapPropsToLocale: (props: any) => string) {
```

Comment:
```suggestion
  return function createHOC<U>(mapPropsToLocale: (props: U) => string) {
```

Reply:
<img width="868" alt="Screen Shot 2019-06-12 at 8 49 12 AM" src="https://user-images.githubusercontent.com/2425013/59329400-1346f300-8cef-11e9-9880-c2057198b017.png">

Reply:
I guess it makes sense because they can set their locale prop as anything 🤔

Reply:
That's why you need to specify the props shape

```ts
type TestProps = {
  locale: string
};

const WrappedComponent = l10nInjector<TestProps>(props => props.locale)(Foo);
```